testutil/testproxy/relayerproxy.go

```diff
@@ -0,0 +1,469 @@
+	secp256k1 "github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/types"
+	accounttypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/golang/mock/gomock"
+	"github.com/noot/ring-go"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/pkg/client"
+	"github.com/pokt-network/poktroll/pkg/relayer"
+	"github.com/pokt-network/poktroll/pkg/relayer/proxy"
+	"github.com/pokt-network/poktroll/pkg/signer"
+	mockclient "github.com/pokt-network/poktroll/testutil/mockclient"
+	mockaccount "github.com/pokt-network/poktroll/testutil/mockrelayer/account"
+	mockapp "github.com/pokt-network/poktroll/testutil/mockrelayer/application"
+	mockkeyring "github.com/pokt-network/poktroll/testutil/mockrelayer/keyring"
+	mocksession "github.com/pokt-network/poktroll/testutil/mockrelayer/session"
+	mocksupplier "github.com/pokt-network/poktroll/testutil/mockrelayer/supplier"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	suppliertypes "github.com/pokt-network/poktroll/x/supplier/types"
+)
+
+// relayerProxyMocks is a struct that holds the mocks for the relayer proxy dependencies
+type relayerProxyMocks struct {
+	blockClientMock     *mockclient.MockBlockClient
+	accountQuerierMock  *mockaccount.MockQueryClient
+	appQuerierMock      *mockapp.MockQueryClient
+	sessionQuerierMock  *mocksession.MockQueryClient
+	supplierQuerierMock *mocksupplier.MockQueryClient
+	keyringMock         *mockkeyring.MockKeyring
+}
+
+// relayerProxyDeps is a struct that holds the dependencies for the relayer proxy
+type relayerProxyDeps struct {
+	clientCtx       relayer.QueryClientContext
+	blockClient     client.BlockClient
+	accountQuerier  accounttypes.QueryClient
+	appQuerier      apptypes.QueryClient
+	sessionQuerier  sessiontypes.QueryClient
+	supplierQuerier suppliertypes.QueryClient
+	keyring         keyringtypes.Keyring
+}
+
+type ProvidedServiceConfig struct {
+	Url     string
+	RpcType sharedtypes.RPCType
+}
+
+// TestBehavior is a struct that holds the test context and mocks
+// for the relayer proxy tests
+type TestBehavior struct {
+	ctx   context.Context
+	t     *testing.T
+	ctrl  *gomock.Controller
+	Deps  depinject.Config
+	mocks *relayerProxyMocks
+
+	proxiedServicesConfig    map[string]string
+	ProxiedServicesEndpoints map[string]*url.URL
+	proxiedServices          map[string]*http.Server
+
+	ProvidedServices map[string]ProvidedServiceConfig
+
+	SupplierKeyName string
+	supplierAddress types.AccAddress
+
+	ApplicationPrivateKey *secp256k1.PrivKey
+}
+
+// ValidPayload is a helper variable for tests that need a valid payload
+var ValidPayload = &servicetypes.JSONRPCRequestPayload{
+	Method:  "someMethod",
+	Id:      1,
+	Jsonrpc: "2.0",
+	Params:  []string{"someParam"},
+}
```

Comment:
Please merge with main
```suggestion
var ValidPayload = []byte(`{"method":"someMethod","id":1,"jsonrpc":"2.0","params":["someParam"]}`)
```
See https://github.com/pokt-network/poktroll/blob/main/pkg/partials/partials_test.go for similar stuff